src/components/vm/nics/vmNicsCard.jsx

```diff
@@ -348,29 +355,45 @@ export class VmNetworkTab extends React.Component {
             },
             {
                 name: _("IP address"),
-                value: () => {
-                    const ips = this.state.ips;
-
-                    if (!Object.keys(ips).length) {
+                value: (network, networkId) => {
+                    const ips = this.state.ips.filter(ip => ip.mac === network.mac);
+
+                    const ip = {};
+                    // If information from agent are available, it's preferred over lease/arp
+                    ips.sort((a, _) => a === "agent" ? -1 : 1).forEach(a => {
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1160-20230725-142141-ae45fd2d-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)